test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/InstanceAnnotations/AnnotationTest.cs

```diff
@@ -0,0 +1,259 @@
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.AspNet.OData.Routing.Conventions;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+using Microsoft.Test.E2E.AspNet.OData.Common.Execution;
+using Microsoft.Test.E2E.AspNet.OData.Common.Extensions;
+using Microsoft.Test.E2E.AspNet.OData.ModelBuilder;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using NS;
+
+namespace Microsoft.Test.E2E.AspNet.OData.InstanceAnnotations
+{
+    public class AnnotationTest : WebHostTestBase
+    {
+        public AnnotationTest(WebHostTestFixture fixture)
+            :base(fixture)
+        {
+        }
+
+        protected override void UpdateConfiguration(WebRouteConfiguration configuration)
+        {
+            var controllers = new[] { typeof(EmployeesController), typeof(MetadataController) };
+            configuration.AddControllers(controllers);
+
+            configuration.Routes.Clear();
+            configuration.Count().Filter().OrderBy().Expand().MaxTop(null).Select();
+            configuration.MapODataServiceRoute("convention", "convention", AnnotationEdmModel.GetConventionModel(configuration));
+            configuration.MapODataServiceRoute("explicit", "explicit", AnnotationEdmModel.GetExplicitModel(configuration), new DefaultODataPathHandler(), ODataRoutingConventions.CreateDefault());
+            configuration.EnsureInitialized();
+        }
+
+        #region InstanceAnnotation
+
+
+
+        [Theory]
+        [InlineData("application/json;odata.metadata=full")]
+        [InlineData("application/json;odata.metadata=minimal")]
+        [InlineData("application/json;odata.metadata=none")]
+        public async Task InstanceAnnotationOnTypeTest(string format)
+        {
+            await ResetDatasource();
+            string postUri = this.BaseAddress + "/convention/Employees?$format="+format;
+
+            var postContent = JObject.Parse(@"{""ID"":1,                                       
+                    ""Name"":""Name2"",
+                    ""SkillSet"":[""Sql""],
+                    ""Gender"":""Female"",
+                    ""AccessLevel"":""Read,Write"",
+                    ""FavoriteSports"":{
+                            ""LikeMost"":""Pingpong"",
+                            ""Like"":[""Pingpong"",""Basketball""]
+                    },
+                    ""@NS.Test"":1}");
+
+            using (HttpResponseMessage response = await this.Client.PostAsJsonAsync(postUri, postContent))
+            {
+                Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+            }
+
+            using (HttpResponseMessage response = await this.Client.GetAsync(postUri))
+            {
+                response.EnsureSuccessStatusCode();
+
+                var json = await response.Content.ReadAsObject<JObject>();
+                var result = json.GetValue("value") as JArray;
+                Assert.Equal<int>(4, result.Count);
+            }
```

Comment:
what tests here?

You only verify the item count, but not verify the instance annotation deseriazliation.

At least,
1)  in the Post([FromBody]Employee employee) method of the controller, to verify the employee.InstanceAnnotations has the value,  for example it should has "NS.Test"/1 pairs
2)  if the response has the instance annotation, should verify the response annotation correct.